src/Adapters/PageCategoriesRetriever.php

```diff
@@ -0,0 +1,22 @@
+<?php
+
+declare( strict_types = 1 );
+
+namespace ProfessionalWiki\PageApprovals\Adapters;
+
+use Title;
+use WikiPage;
+
+class PageCategoriesRetriever {
+
+	/**
+	 * @return string[]
+	 */
+	public function getPageCategories( WikiPage $page ): array {
+		return array_map(
+			fn( Title $categoryTitle ) => $categoryTitle->getText(),
+			iterator_to_array( $page->getCategories() )
+		);
+	}
```

Comment:
Is this logic too trivial to have a dedicated service? The same logic is used elsewhere: https://github.com/ProfessionalWiki/PageApprovals/pull/43/files#diff-e71997fafbe96a8d4e96646051b2c13fae74c6b8af4b4cec0f1e72e302bdd0b8R45-R50

So a service would deduplicate that while having tests.

Reply:
Would it be better if we make Categories matching case insensitive? I ran into errors while testing this and took me quite some time to figure out that my case was not right.

Reply:
Categories are not fully case-insensitive:
* `Category:Foo` = `Category:foo`
* `Category:Some thing` != `Category:Some Thing`

The first letter is always case insensitive, but any other letter is not.

Reply:
![image](https://github.com/ProfessionalWiki/PageApprovals/assets/15813104/6eb07103-1b4e-460f-82a2-2626c04c4d09)

Its not working for me.

Reply:
MediaWiki already has normalization for titles. We should use that, if it is not already used, and then verify.

Reply:
@malberts seems silly to have a service for that, but that mostly comes from such a service not being proper OO due to how MediaWiki is doing things. The method really should be on WikiPage, or on it's return value (TitleArray?) like `$wikiPage->getCategories()->asStringArray()`

Reply:
I was thinking that `PageCategoriesRetriever` might provide value in making tests easier, but that does not seem the case either. Having this service does not seem that harmful, but to be it looks slightly better to not have it.

Reply:
@JeroenDeDauw I removed the class.